core/runtime.rs

```diff
@@ -1213,7 +1226,17 @@ impl JsRuntime {
     // Update status after evaluating.
     status = module.get_status();
 
-    if let Some(value) = maybe_value {
+    let has_explicit_terminate_error =
+      state_rc.borrow().explicit_terminate_error.is_some();
+
+    if has_explicit_terminate_error {
+      let undefined = v8::undefined(tc_scope);
+      // Pass `undefined`, it will be substituted with
+      // `explicit_terminate_error` within `exception_to_err_result()`.
+      sender
+        .send(exception_to_err_result(tc_scope, undefined.into(), false))
+        .expect("Failed to send module evaluation error.");
+    } else if let Some(value) = maybe_value {
```

Comment:
While this works, why not take the `explicit_terminate_error` here and pass it explicitly instead of taking it in `exception_to_error_result`?

Reply:
I need it to be available inside of the fn to avoid the code path outside of `if let Some() = explicit_terminate_error.take()`. Also I got a weird serde_v8 panic if I didn't convert `exception -> js_error` ahead of time.

Reply:
Okay, that's fine. Could you add a comment about it? Seems like a potential future refactor